lib/config/validation.ts

```diff
@@ -147,9 +147,13 @@ export async function validateConfig(
         message: `The "${key}" object can only be configured at the top level of a config but was found inside "${parentPath}"`,
       });
     }
-    if (!isGlobalConfig) {
+
+    if (isGlobalConfig) {
+      validateGlobalConfig(key, val, optionTypes[key], warnings, currentPath);
```

Comment:
```suggestion
    if (isGlobalConfig && validateGlobalConfig(key, val, optionTypes[key], warnings, currentPath)) {
```

Idea, `validateGlobalConfig` should return true when it validated the `key` and false when not, so it can fall through below to validate other options

or only call it when `key` is a global option